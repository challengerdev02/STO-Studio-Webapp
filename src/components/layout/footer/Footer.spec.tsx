import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { Footer } from './index';
import Logo from '../../../../public/assets/logo-dark.svg';

describe('Footer', function () {
  afterEach(cleanup);

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<Footer imgSrc={Logo} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should test the footer is rendered with the right texts', function () {
    const { getByText } = setup();
    expect(getByText('For Creators')).toBeInTheDocument();
    expect(getByText('Connect wallet')).toBeInTheDocument();
    expect(getByText('Accept')).toBeInTheDocument();
  });
});
