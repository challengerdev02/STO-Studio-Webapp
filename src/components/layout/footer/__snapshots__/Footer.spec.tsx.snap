// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match with snapshot 1`] = `
<div>
  <footer
    class="sc-eCImPb hEiTOE"
  >
    <div
      class="sc-crHmcD hqInhG"
    >
      <div
        class="sc-egiyK gUTRyv"
      >
        <div
          class="sc-iqseJM boshHq"
        >
          <img
            alt="logo"
            class="sc-jRQBWg ljlGsw"
            src="/test-file-stub"
          />
          <div
            class="sc-bdvvtL ZBEGP"
          >
            MetaComic
          </div>
        </div>
        <div
          class="sc-gKclnd jntxQG"
        >
          For Creators
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal sc-dkPtRN fWmImA"
        role="separator"
      />
      <div
        class="sc-hKwDye jzumDG"
      >
        MetaComic
        <div
          class="sc-iCfMLu iJNXsJ"
        >
          <div
            class="sc-pVTFL jCInJp"
          >
            Explore
          </div>
          <div
            class="sc-pVTFL jCInJp"
          >
            Connect wallet
          </div>
          <div
            class="sc-pVTFL jCInJp"
          >
            Create item
          </div>
        </div>
      </div>
      <div
        class="ant-divider ant-divider-horizontal sc-dkPtRN fWmImA"
        role="separator"
      />
      <div
        class="sc-hKwDye jzumDG last"
      >
        Community
        <div
          class="sc-iCfMLu iJNXsJ"
        >
          <div
            class="sc-pVTFL jCInJp"
          >
            MetaComic Token
          </div>
          <div
            class="sc-pVTFL jCInJp"
          >
            Discussion
          </div>
          <div
            class="sc-pVTFL jCInJp"
          >
            Voting
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-divider ant-divider-horizontal sc-gsDKAQ bEvupN"
      role="separator"
    />
    <div
      class="sc-furwcr choPSI"
    >
      <div
        class="sc-jrQzAO iChvoW"
      >
        Copyright Â© 2022 MetaComic. All rights reserved
      </div>
      <div
        class="sc-bqiRlB jUHPkE"
      >
        <div
          class="sc-kDTinF cJnHAq"
        >
          We use cookies for better service.
          <span>
            <button
              class="ant-btn ant-btn-link"
              style="padding-right: 0px;"
              type="button"
            >
              <span>
                Accept
              </span>
            </button>
          </span>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
